Apps/W1/HybridGP/app/src/Migration/History/GPPopulateHistTables.Codeunit.al

```diff
@@ -0,0 +1,957 @@
+    begin
+        if GPCompanyAdditionalSettings.GetMigrateHistory() then begin
+            PopulateGLDetail();
+            PopulateReceivables();
+            PopulatePayablesDocuments();
+            PopulateInventoryTransactions();
+            PopulatePurchaseRecvTransactions();
+            HistMigrationStatusMgmt.SetStatusFinished();
+        end;
+    end;
+
+    local procedure PopulateGLDetail()
+    begin
+        // GP G/L Accounts
+        if HistMigrationStatusMgmt.HasNotRanStep("Hist. Migration Step Type"::"GP GL Accounts") then
+            if GPCompanyAdditionalSettings.GetMigrateHistGLTrx() then begin
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Accounts", false);
+                PopulateHistGLAccounts();
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Accounts", true);
+            end;
+
+        // GP G/L Journal Trx.
+        if HistMigrationStatusMgmt.HasNotRanStep("Hist. Migration Step Type"::"GP GL Journal Trx.") then
+            if GPCompanyAdditionalSettings.GetMigrateHistGLTrx() then begin
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Journal Trx.", false);
+                PopulateOpenYearPostedTransactions();
+                PopulateHistoricalYearPostedTransactions();
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Journal Trx.", true);
+            end;
+    end;
+
+    local procedure PopulateHistGLAccounts()
+    var
+        GPGL00105: Record "GP GL00105";
+        GPGL00100: Record "GP GL00100";
+        HistGLAccount: Record "Hist. G/L Account";
+    begin
+        if GPGL00105.FindSet() then
+            repeat
+                Clear(HistGLAccount);
+                if GPGL00100.Get(GPGL00105.ACTINDX) then begin
+                    HistGLAccount."No." := CopyStr(Format(GPGL00105.ACTNUMST), 1, MaxStrLen(HistGLAccount."No."));
+                    HistGLAccount.Name := GPGL00100.ACTDESCR;
+
+                    if not HistGLAccount.Insert() then
+                        HistMigrationStatusMgmt.ReportError("Hist. Migration Step Type"::"GP GL Accounts", GPGL00105.ACTNUMST,
+                            GetLastErrorCode(), GetLastErrorText());
+                end;
+            until GPGL00105.Next() = 0;
+    end;
+
+    local procedure PopulateOpenYearPostedTransactions()
+    var
+        GPGL20000: Record "GP GL20000";
+        GPGL00105: Record "GP GL00105";
+        HistGenJournalLine: Record "Hist. Gen. Journal Line";
+        OutlookSynchTypeConv: Codeunit "Outlook Synch. Type Conv";
+        InitialHistYear: Integer;
+    begin
+        InitialHistYear := GPCompanyAdditionalSettings.GetHistInitialYear();
+        if InitialHistYear > 0 then
+            GPGL20000.SetFilter(OPENYEAR, '>=%1', InitialHistYear);
+
+        if GPGL20000.FindSet() then
+            repeat
+                Clear(HistGenJournalLine);
```

Comment:
Maybe we should move this one within the IF statement?

Reply:
Updated